<a href="https://colab.research.google.com/github/sandeepangh782/Analysis-of-Spotify-Dataset/blob/main/Descriptive%20Statistics%20Analysis%20of%20Suicide%20Rate%20from%201985%20to%202016ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**About the dataset:**This compiled dataset pulled from four other datasets linked by time and place, and was built to find signals correlated to increased suicide rates among different cohorts globally, across the socio-economic spectrum.

**Problem Statement:** The suicide rate in lots of countries have been quite high for many years, including the developing and developed countries. At the begining, we'll find out the key factors of high suicide rate.

In [1]:
from google.colab import drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


In [3]:
import numpy as np
import pandas as pd

In [4]:
df = pd.read_csv(r'/content/gdrive/MyDrive/Multivariate/master.csv')

displays the first 10 rows of the DataFrame

In [5]:
df.head()

country  year     sex          age  suicides_no  population  \
0  Albania  1987    male  15-24 years           21      312900   
1  Albania  1987    male  35-54 years           16      308000   
2  Albania  1987  female  15-24 years           14      289700   
3  Albania  1987    male    75+ years            1       21800   
4  Albania  1987    male  25-34 years            9      274300   

   suicides/100k pop country-year  HDI for year  gdp_for_year ($)   \
0               6.71  Albania1987           NaN      2,156,624,900   
1               5.19  Albania1987           NaN      2,156,624,900   
2               4.83  Albania1987           NaN      2,156,624,900   
3               4.59  Albania1987           NaN      2,156,624,900   
4               3.28  Albania1987           NaN      2,156,624,900   

   gdp_per_capita ($)       generation  
0                 796     Generation X  
1                 796           Silent  
2                 796     Generation X  
3                 796  G.I. Generation  
4                 796          Boomers

Provides the statistical information of the dataframe.

count: It indicates the number of data points available for analysis in each column.

mean: It represents the average value of the data in that column.

std: It measures the amount of variation or dispersion in the data points around the mean.

min: It shows the smallest value present in the data.

25%: It represents the value below which 25% of the data falls.

50%: It represents the value below which 50% of the data falls.

75%: It represents the value below which 75% of the data falls.

max: It shows the largest value present in the data.

In [58]:
df.describe()

year   suicides_no    population  suicides/100k pop  \
count  27820.000000  27820.000000  2.782000e+04       27820.000000   
mean    2001.258375    242.574407  1.844794e+06          12.816097   
std        8.469055    902.047917  3.911779e+06          18.961511   
min     1985.000000      0.000000  2.780000e+02           0.000000   
25%     1995.000000      3.000000  9.749850e+04           0.920000   
50%     2002.000000     25.000000  4.301500e+05           5.990000   
75%     2008.000000    131.000000  1.486143e+06          16.620000   
max     2016.000000  22338.000000  4.380521e+07         224.970000   

       HDI for year  gdp_per_capita ($)  
count   8364.000000        27820.000000  
mean       0.776601        16866.464414  
std        0.093367        18887.576472  
min        0.483000          251.000000  
25%        0.713000         3447.000000  
50%        0.779000         9372.000000  
75%        0.855000        24874.000000  
max        0.944000       126352.000000

Mean suicide rates from year 1985 t0 2016

In [40]:
mean_suicides = df.groupby(['year'])['suicides_no'].count()
print(mean_suicides)
print("Average suicide number",mean_suicides.mean().round())

year
1985     576
1986     576
1987     648
1988     588
1989     624
1990     768
1991     768
1992     780
1993     780
1994     816
1995     936
1996     924
1997     924
1998     948
1999     996
2000    1032
2001    1056
2002    1032
2003    1032
2004    1008
2005    1008
2006    1020
2007    1032
2008    1020
2009    1068
2010    1056
2011    1032
2012     972
2013     960
2014     936
2015     744
2016     160
Name: suicides_no, dtype: int64
Average suicide number 869.0


In the given duration the average suicide rate was around 869 people. In the
2009 the suicide rate was the highest

Median suicide rate from 1985 to 2016

In [41]:
median_suicides = df.groupby(['year'])['suicides_no'].count()
print(mean_suicides)
print("The median of suicide rates:",mean_suicides.median().round())

year
1985     576
1986     576
1987     648
1988     588
1989     624
1990     768
1991     768
1992     780
1993     780
1994     816
1995     936
1996     924
1997     924
1998     948
1999     996
2000    1032
2001    1056
2002    1032
2003    1032
2004    1008
2005    1008
2006    1020
2007    1032
2008    1020
2009    1068
2010    1056
2011    1032
2012     972
2013     960
2014     936
2015     744
2016     160
Name: suicides_no, dtype: int64
The median of suicide rates: 942.0


The median of the suicide rate is 942

In [50]:
sex = df.groupby(['year','sex'])['suicides_no'].sum()
print(sex)

year  sex   
1985  female     32479
      male       83584
1986  female     33852
      male       86818
1987  female     35006
                 ...  
2014  male      171428
2015  female     47248
      male      156392
2016  female      3504
      male       12099
Name: suicides_no, Length: 64, dtype: int64


Mode of the suicide numbers

In [63]:
col_list = df["suicides_no"].mode()
print(col_list)

0    0
Name: suicides_no, dtype: int64


Standard deviation in suicide rate per 100k population column.

In [65]:
std_dev_suicides_100k = df['suicides/100k pop'].std().round()
print(f"Standard deviation of suicides/100k pop: {std_dev_suicides_100k}")

Standard deviation of suicides/100k pop: 19.0


Correaltion between suicide numbers and population

In [66]:
correlation_suicides_population = df['suicides_no'].corr(df['population'])
print(f"Correlation between suicides_no and population: {correlation_suicides_population}")

Correlation between suicides_no and population: 0.6161622675219285


From the correlation the output is 0.6 which is positive correaltion therefore we can say that with increate in population there is increase in suicide rates.

**Skewness**

In [67]:
skewness_suicides = df['suicides_no'].skew()
print(f"Skewness of suicides_no: {skewness_suicides}")

Skewness of suicides_no: 10.352910335856805


Positively skewed which means most of the suicide rates lie in the right of the mean.

Kurtosis

In [68]:
kurtosis_suicides = df['suicides_no'].kurt()
print(f"Kurtosis of suicides_no: {kurtosis_suicides}")

Kurtosis of suicides_no: 157.1688417742334


kurtosis values are typically around 3 for a normal distribution. So, a kurtosis value of 157.1688 is extremely high and indicates an extremely heavy-tailed distribution with a very sharp peak. A high kurtosis value could be an indication of outliers or a non-standard distribution in the dataset

In [69]:
covariance_suicides_population = df['suicides_no'].cov(df['population'])
print(f"Covariance between suicides_no and population: {covariance_suicides_population}")

Covariance between suicides_no and population: 2174197877.0798373


the covariance value of 2174197877.0798373 suggests that there is some relationship between "suicides_no" and "population", as the "suicides_no" variable increases, the "population" variable tends to increase as well. It indicates that in the given dataset, larger populations are associated with higher suicide numbers. The large covariance value may o indicate a strong relationship between the two variables as if the "suicides_no" and "population" values are large, the covariance will also be large, even if the relationship is not particularly strong.

Correlation matrix

In [70]:
correlation_matrix = df.corr()
print("Correlation matrix:")
print(correlation_matrix)

Correlation matrix:
                        year  suicides_no  population  suicides/100k pop  \
year                1.000000    -0.004546    0.008850          -0.039037   
suicides_no        -0.004546     1.000000    0.616162           0.306604   
population          0.008850     0.616162    1.000000           0.008285   
suicides/100k pop  -0.039037     0.306604    0.008285           1.000000   
HDI for year        0.366786     0.151399    0.102943           0.074279   
gdp_per_capita ($)  0.339134     0.061330    0.081510           0.001785   

                    HDI for year  gdp_per_capita ($)  
year                    0.366786            0.339134  
suicides_no             0.151399            0.061330  
population              0.102943            0.081510  
suicides/100k pop       0.074279            0.001785  
HDI for year            1.000000            0.771228  
gdp_per_capita ($)      0.771228            1.000000  


<ipython-input-70-7a7341cd5616>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  correlation_matrix = df.corr()


The correlation between suicides and the GDP per capita is also positive but weak (0.061). This indicates that there might be a slight positive association between a country's economic prosperity (higher GDP per capita) and the number of suicides, but again, the correlation is not strong, so it may be one of the factors affecting suicide rates

The correlation between suicides and population is moderate (0.616). This suggests that there is a moderate positive association between a larger population and a higher number of suicides.

The correlation between the year and suicides per 100k population is negative (-0.039). This indicates that the suicide rate per 100,000 people might have a slight downward trend over the years, we can see that after 2015 there is a downward trend in suicide rates even though there had been rise in the rates before.